Oasis-II

Freelanci opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->

A modernization of Oasis station, with a lower minimum player count required for it to show up on map votes.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->

Awhile ago, a vote was held on the server to force the map to Oasis, out of curiosity as many of us have never seen the map. Despite being nowhere near the required minimum player count to even have access to the map normally (70 players), the round went quite well with our usual player counts. 

A desire to include a version of Oasis that had a much lower minimum player limit (around 35 players) was made apparent. Due to us never previously hitting the required player count to even vote for Oasis, there is justification to make a ronfeature version of this map despite the fact it is still maintained upstream. The map is old, and could use adjustments besides simply lowering the minimum player count.

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

Map render coming soon. Anticipating problems with the render due to the separate perma exile grid. A workaround is planned, but will take some effort to pull off.

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->

:cl: Freelanci, hiu(ophiuchiko), BlueVibes, Timtee_CVHN, IszAku, oliverwf
MAPS:
- add: On Oasis-2: Brigmed has been added to the security department, taking the place of an abandoned maints surgery room that was practically begging to be remade into brigmed. This room was largely designed and built by hiu (ophiuchiko).
- add: On Oasis-2: Added a new escape pod south of the arcade.
- add: On Oasis-2: Added many new station beacons to relevant rooms, and renamed existing beacons that were excessively long.
- add: On Oasis-2: Added a roundstart supply bot to cargo, because why not? This ghost role hardly ever gets use despite it being buffed not too long ago.
- add: On Oasis-2: A new ghost role in the anomaly lab; Jorge. He is a sentient potted eyeball plant that can do some complex interactions, making him somewhat useful to have around. He can move on his own, albeit quite slowly, and can be somewhat difficult to understand when he speaks.
- add: On Oasis-2: A new salon to take up room left over from the TEG rework.
- add: On Oasis-2: A BSO office along with a corresponding job slot for BSO.
- remove: On Oasis-2: Removed copious amounts of hardened reinforced floor from various parts of the station. This type of flooring cannot be removed by players without the use of explosives, and thus should be used sparingly.
- remove: On Oasis-2: Removed air alarms from maints. These shouldn't have been there in the first place. Engineers need things to do like installing their own air alarms in these locations.
- remove: On Oasis-2: A great purge of unnecessary air sensors and fire alarms has taken place, noticeably de-cluttering the walls and floors of the station.
- tweak: On Oasis-2: The courtyard (giant park in the center) received a substantial aesthetic rework, primarily thanks to hiu (ophiuchiko). In this change, the zoo pens have been made a lot more varied - and secured, in the case of the gorilla cage. Additionally, a barn section has been added to the northeast quadrant so the kitchen can have access to milk and eggs.
- tweak: On Oasis-2: The existing dispo/mail chute systems were uprooted and replaced with a new hybrid chute system. This new system has security features which prevent two different forms of disposals rerouting into secure departments (sec, command, engi).
- tweak: On Oasis-2: The antiquated TEG design on oasis was completely replaced with a modern equivalent. Current estimated difficulty/time spent on TEG setup is roughly equivalent to that of Plasma station. Credit to Timtee for designing this TEG.
- tweak: On Oasis-2: The PA chamber received a much needed upgrade. In the original oasis, the PA chamber was more incomplete than the chamber on elkridge, and had mixed tesla/singulo components messily scattered around the chamber. It also originally started with MOST of the MV cables missing. The chamber now adopts color-coded decals - inspired by Exo station - which denote machine configurations for singulo (purple) and tesla (blue). Like the original Oasis, the PA chamber is configured for singulo by default.
- tweak: On Oasis-2: While it was extremely humorous, the AME has been moved to a different room slightly further away from the captain's office. This was done to make room for a BSO office that would take its place (currently WIP).
- tweak: On Oasis-2: Evac docks and the evac shuttle have been adjusted. Meteor shielding is now in place around the evac docks to prevent the room from being destroyed prior to round end. The shuttle had sections that were far too easy for people to space, as they only had a directional reinforced window between the atmosphere and space. These windows were replaced with shuttle windows. Finally, evac docks have been redecorated, with a lot of inspiration coming from the evac dock of Core station.
- tweak: On Oasis-2: Several sections of the security department have been given their own APC. This includes the armory. The HV lines electrifying the grilles around armory have been given their own dedicated SMES, to prevent surges to the whole station grid if it shocks something.
- tweak: On Oasis-2: The comically insecure directional windows used for genpop visitation have been replaced with a full window block paired with an MV electrified grille. Genpop prisoners get shovels on this station, a couple directional windows are not going to cut it.
- tweak: On Oasis-2: The firing range south of security has been fitted with a new conveyor system to allow for moving targets. Thanks to hiu (ophiuchiko) for this change.
- tweak: On Oasis-2: Main hallways around Oasis, apart from the central courtyard have had tile/decal rework inspired by a design seen in the old Origin station. Thanks to hiu (ophiuchiko) and iszaku (aku) for help with this.
- tweak: On Oasis-2: Anomaly lab has taken the place of where artifact lab used to be (next to the hallway shared between sci and medbay. The artifact chambers were very small, and in a horrible position, as several scenarios could result in the medbay hallway being spaced. Artlab has been moved south of sci, where the "munitions testing lab" used to be. This artlab features two 3x3 chambers with many bells and whistles. Credit for ideas on artlab is to be given to oliverfw. A gas can storage room has taken the place of anomaly lab's old position.
- tweak: On Oasis-2: The offices of RD, CMO, QM, CE and HoS now have their own APCs, preventing trivial break-ins by de-powering the rooms from an external APC.
- tweak: On Oasis-2: Shutters leading to various rooms (paramedic, robotics, janitor) have been backed with department-locked secure windoors to prevent easy intrusion.
- tweak: On Oasis-2: Externally-facing airlock pairs have been made resilient to the effects of the GreyTide virus event, preventing mass spacings.
- tweak: On Oasis-2: As a result of removing hardened reinforced floors from around the station, the vault has received a functional and aesthetic rework. The golden toolbox - originally missing - has been added to the vault. A sentry turret complete with with internal and external control panels has been added to the vault.
- tweak: On Oasis-2: For added security, the vault and tech vault have a dedicated SMES and substation positioned between the two rooms.
- tweak: On Oasis-2: Security and command comms servers have been relocated to the tech vault, while the original comms room got an aesthetic rework. Similarly, security and command camera routers have been relocated to the AI lobby.
- tweak: On Oasis-2: Virology was reduced in size to make room for the gravity generator's new position (south of the comms room).
- tweak: On Oasis-2: The complex mess of logic gates around the salvage dock were reduced to two gates. This mess of logic gates appeared to be the original mapper's attempt at avoiding the use of directional fans in a certain position. Needless to say, I couldn't justify replicating their work, and decided to go with the directional fans.
- tweak: On Oasis-2: Improved the medbay cryo pod setup. Added filters for CO2, N2O and ammonia, and made it so theses gases are sent to a waste line instead of a canister.
- tweak: On Oasis-2: Many directional windows and windoors in medbay examination have been removed, particularly those close to medical beds. These placements are often found to be restrictive by players.
- tweak: On Oasis-2: Chemistry now uses the correct color scheme regarding the decals on the floors.
- tweak: On Oasis-2: Improved the controls of the disposals conveyors.
- tweak: On Oasis-2: All asteroid walls have been replaced with sandstone walls for greater aesthetic appeal.
- tweak: On Oasis-2: Improved meteor shielding around atmos.
- tweak: On Oasis-2: The tech vault has been reworked to be smaller and less cluttered. The remaining space has been converted into an engineering materials closet.
- tweak: On Oasis-2: AI core has been modernized. Turrets added, moderate redecoration.

imatsoup commented:
Typically if you're still working on a pull request you're opening, you'll want to set it up as a draft.

Freelanci commented:
> Typically if you're still working on a pull request you're opening, you'll want to set it up as a draft.

Done, converted to draft.

arenaconspiracy commented:
Removing the DNM label because the draft status makes it redundant.
I'll try to take a look today if I can but I will likely be away from home for a long time, so that might not happen.

arenaconspiracy commented on Resources/Prototypes/_Ronstation/Maps/oasis-2.yml:
This file is in our _Ronstation subdirectory and therefore doesn't need these headers. YML isn't even code and is hardly subject to this anyway.

imatsoup replied:
YML outside of our own subdirectory would be subject to this, but yeah, we dont need the header or the #Ronstation comments for files in our own subdirectories

imatsoup commented on Content.IntegrationTests/Tests/PostMapInitTest.cs:
```diff
@@ -71,6 +71,7 @@ public sealed class PostMapInitTest
             "Saltern",
             "Reach",
             "Oasis",
+			"Oasis2",
```
```suggestion
			"Oasis2", # Ronstation - Modification
```

Freelanci replied:
Wait a minute, this is cs. Is this the correct syntax for a comment?

arenaconspiracy commented on Resources/Prototypes/_Ronstation/Maps/oasis-2.yml:
```diff
@@ -0,0 +1,68 @@
+# Contains modifications made by Ronstation contributors, therefore this file is subject to MIT sublicensed with AGPL v3.0.
```
Doesn't need license comments because it's in our subdirectory.

Freelanci commented:
How would we go about giving people attributions?

Freelanci commented:
Besides attributions, we should be good to go.

arenaconspiracy commented:
That is not what I meant.
This is the way to do it, mocked up using element inspector. 
<img width="531" height="293" alt="tmp" src="https://github.com/user-attachments/assets/02076d89-2acf-46c2-93eb-789bda0efe30" />

Freelanci commented:
Done

Freelanci commented:
I just caught a mistake I will be patching shortly. When I removed the sledgehammer from robotics (the one that had 2x the normal damage), I replaced it with a joke version, with sqeak-toy sound effects and no damage. I failed to account for the increased damage upon wielding, which returns its damage to 10 blunt and 10 structural. To be honest, this is still better than the buffed one in the original oasis, but I still want to make that sledgehammer entirely a toy.

arenaconspiracy commented:
The normal sledgehammer isn't buffed it just has IncreaseDamageOnWieldComponent.
I ask that instead of a useless joke item nobody will ever look at you either remove the sledgehammer entirely, or replace it with something reasonable like a baseball bat + crowbar.

Freelanci commented:
There, removed and replaced by a crowbar

arenaconspiracy reviewed (approved):
Latest go around works. Caught everything major in reviews, leaving the rest to trial forum.
